packages/unmock-core/src/interceptor/index.ts

```diff
@@ -0,0 +1,34 @@
+import { ISerializedRequest, ISerializedResponse } from "../interfaces";
+import NodeInterceptorConstructor from "./node-interceptor";
+
+export interface IInterceptorListener {
+  createResponse(request: ISerializedRequest): ISerializedResponse | undefined;
+}
+
+export interface IInterceptorOptions {
+  listener: IInterceptorListener;
+  shouldBypassHost: (host: string) => boolean;
+}
+
+/**
+ * Create interceptor and start intercepting requests.
+ * @param options
+ */
+export type IInterceptorConstructor = new (
+  options: IInterceptorOptions,
+) => IInterceptor;
+
+/**
+ * Active interceptor.
+ */
+export interface IInterceptor {
+  disable(): void;
+}
+
+/**
+ * Create an interceptor and start intercepting requests.
+ * @param options
+ */
+export function createInterceptor(options: IInterceptorOptions): IInterceptor {
```

Comment:
Is this here for the purpose of being a future factory function, or encapsulating the `NodeInterceptorConstructor`?

Reply:
Yes it's a future factory function, possibly dynamically creating a suitable interceptor (Node.js vs browser) based on the current environment.